src/ui/public/utils/brush_event.js

```diff
@@ -9,43 +9,55 @@ export function UtilsBrushEventProvider(timefilter) {
         return;
       }
 
-      switch (event.data.xAxisField.type) {
-        case 'date':
-          const from = moment(event.range[0]);
-          const to = moment(event.range[1]);
-
-          if (to - from === 0) return;
-
-          timefilter.time.from = from;
-          timefilter.time.to = to;
-          timefilter.time.mode = 'absolute';
-          break;
-
-        case 'number':
-          if (event.range.length <= 1) return;
-
-          const existingFilter = $state.filters.find(filter => (
-            filter.meta && filter.meta.key === event.data.xAxisField.name
-          ));
-
-          const min = event.range[0];
-          const max = event.range[event.range.length - 1];
-          const range = { gte: min, lt: max };
-          if (_.has(existingFilter, 'range')) {
-            existingFilter.range[event.data.xAxisField.name] = range;
-          } else if (_.has(existingFilter, 'script.script.params.gte')
-            && _.has(existingFilter, 'script.script.params.lt')) {
-            existingFilter.script.script.params.gte = min;
-            existingFilter.script.script.params.lt = max;
-          } else {
-            const newFilter = buildRangeFilter(
-              event.data.xAxisField,
-              range,
-              event.data.indexPattern,
-              event.data.xAxisFormatter);
-            $state.$newFilters = [newFilter];
-          }
-          break;
+      if (event.data.xAxisField.type === 'date' &&
+        event.data.xAxisField.name === event.data.indexPattern.timeFieldName) {
+        setTimefilter();
+      } else if (event.data.xAxisField.type === 'date' || event.data.xAxisField.type === 'number') {
+        setRange();
+      }
+
+      function setTimefilter() {
+        const from = moment(event.range[0]);
+        const to = moment(event.range[1]);
+
+        if (to - from === 0) return;
+
+        timefilter.time.from = from;
+        timefilter.time.to = to;
+        timefilter.time.mode = 'absolute';
+      }
+
+      function setRange() {
+        if (event.range.length <= 1) return;
+
+        const existingFilter = $state.filters.find(filter => (
+          filter.meta && filter.meta.key === event.data.xAxisField.name
+        ));
+
+        let min = event.range[0];
+        let max = event.range[event.range.length - 1];
+        // Convert Dates to MS to avoid ES "parse date field" errors
+        if (min instanceof Date) {
+          min = min.getTime();
+        }
+        if (max instanceof Date) {
+          max = max.getTime();
+        }
+        const range = { gte: min, lt: max };
```

Comment:
I'm still getting the same error. You'll need to change this line to the following:
```javascript
const range = { gte: min, lt: max, format: 'epoch_millis' };
```
Then it will work 😄